service/src/main/java/bio/terra/catalog/service/dataset/DatasetAccessLevel.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.catalog.service.dataset;
+
+import bio.terra.catalog.iam.SamAction;
+import java.util.EnumSet;
+import java.util.Set;
+
+public enum DatasetAccessLevel {
+  OWNER("owner", EnumSet.allOf(SamAction.class)),
+  READER("reader", EnumSet.of(SamAction.READ_ANY_METADATA)),
+  DISCOVERER("discoverer", Set.of());
```

Comment:
I'm not sure this is correct (although it is what the current code is doing). I think `DISCOVERER` can also `READ_ANY_METADATA` (given the way this action is used used in our code, which doesn't quite match the action name).

If we did a permission check for `export` or `preview` operations, then we'd need another action for those operations. Right now those permission checks are done in the storage system services.